go-hello-nonroot/app.go

```diff
@@ -0,0 +1,87 @@
+package main
+
+import (
+	"crypto/rand"
+	"errors"
+	"fmt"
+	"net"
+	"net/http"
+	"os/user"
+	"time"
+)
+
+var id string
+
+func main() {
+	id = generateID()
+	http.HandleFunc("/coffee", cafeHandler)
+	http.HandleFunc("/tea", cafeHandler)
+	http.ListenAndServe(":8080", nil)
+}
+
+func generateID() string {
```

Comment:
I think `generateID()`, `cafeHandler()`, etc should return an error and exit with error code from `main()` instead of just printing the error message and continuing to run.

Something like: 
`log.Fatalf(err)`